pkg/header/injector.go

```diff
@@ -39,6 +40,26 @@ func NewInjector(headers []options.Header) (Injector, error) {
 	return &injector{valueInjectors: injectors}, nil
 }
 
+func updateHeaderWithClaimValues(header *http.Header, name string, claimValues []string, prefix string) {
+	if len(claimValues) == 0 {
+		return
+	}
+	nonEmptyValues := make([]string, 0)
+	existingValues := header.Values(name)
+	if len(existingValues) > 0 {
+		nonEmptyValues = append(nonEmptyValues, strings.Join(existingValues, ","))
+	}
+	for _, claim := range claimValues {
+		if claim != "" {
+			nonEmptyValues = append(nonEmptyValues, prefix+claim)
+		}
+	}
+	if len(nonEmptyValues) > 0 {
+		header.Set(name, strings.Join(nonEmptyValues, ","))
+	}
+
+}
+
```

Comment:
Would this be simpler if instead of making this add to the existing values, we just had the logic in a slightly different place and just flattened the values?

I was thinking we would have something like this in the middleware package that we could call after the header injection

```
func flattenHeaders(headers http.Header) {
  for name, values := range headers {
    headers.Set(name, strings.Join(values, ",")
  }
}
```

This could be called in the injectRequest and injectResponse middlewares after the headers have been injected
Requests:
https://github.com/oauth2-proxy/oauth2-proxy/blob/8e02fac2cc0ae61552d3244b313eb8e0dcd2d545/pkg/middleware/headers.go#L68
Response:
https://github.com/oauth2-proxy/oauth2-proxy/blob/8e02fac2cc0ae61552d3244b313eb8e0dcd2d545/pkg/middleware/headers.go#L99